docs/current_docs/quickstart/refactor.mdx

```diff
@@ -0,0 +1,84 @@
+- It calls the `node().install()` function, which runs `npm install` in the container and returns a revised container with the application's dependencies installed.
+- It calls the `node().commands().run()` function with an array of arguments, which returns a revised container configured to execute the `npm run test:unit run` or the `npm run build` command in the container image.
+- It uses the `container().stdout()` method to return the test results or the `container().directory()` method to return the build output directory.
+
+</TabItem>
+</Tabs>
+
+Let's backtrack a little further and inspect the [Node module's source code](https://github.com/dagger/dagger/blob/9e59bae142f64975b7c9ad851e6bd4901d43513a/sdk/typescript/dev/node/src/index.ts) to see what's happening there.
+
+- The first thing to notice is that the Node module is written in TypeScript. However, its functions are being called from your primary Dagger module, which could be written in Go, Python or TypeScript. Dagger takes care of the translation via its language-agnostic GraphQL layer.
+- The [Node module's constructor](https://github.com/dagger/dagger/blob/9e59bae142f64975b7c9ad851e6bd4901d43513a/sdk/typescript/dev/node/src/index.ts#L22) uses the `container().from()` method of the Dagger API to initialize a new container from a `node` base image, using the version number passed to it. This method returns a `Container` type representing an OCI-compatible container image.
+- The [Node module's `withNpm()` function](https://github.com/dagger/dagger/blob/9e59bae142f64975b7c9ad851e6bd4901d43513a/sdk/typescript/dev/node/src/index.ts#L73) uses the `container().withEntrypoint()` method of the Dagger API to set the container entrypoint to the `npm` CLI and initalizes a cache volume for the `npm` cache using the `container().withMountedCache()` method.
+- The [Node module's `withSource()` function](https://github.com/dagger/dagger/blob/9e59bae142f64975b7c9ad851e6bd4901d43513a/sdk/typescript/dev/node/src/index.ts#L34) uses the `container().withDirectory()`, `container().withWorkdir()` and `container().withMountedCache()` methods of the Dagger API to mount the source code directory in the container, set the working directory to that mount point, and create a cache volume for dependencies.
+- The [Node module's `install()` function](https://github.com/dagger/dagger/blob/9e59bae142f64975b7c9ad851e6bd4901d43513a/sdk/typescript/dev/node/src/index.ts#L109) uses the `container().withExec()` method to define the command to download dependencies in the container (`npm install`).
+- The [Node module's `commands().run()`](https://github.com/dagger/dagger/blob/9e59bae142f64975b7c9ad851e6bd4901d43513a/sdk/typescript/dev/node/src/commands.ts#L13) function accepts an array of command line arguments and readies them for execution in the container, also using the `container().withExec()` method.
+
+Using Dagger Functions from the Node module instead of the core Dagger API offers some advantages:
```

Comment:
IMO this is not a very strong example or reason to use a Dagger module. It has almost the same lines of code as the previous version. I'm looking for ideas to make this example better or show value from refactoring with the module. This could also be a prompt to improve the module itself and add more functions to it. 
Ping @TomChv for thoughts as well.